sqlglot/dialects/mysql.py

```diff
@@ -408,6 +408,41 @@ class Parser(parser.Parser):
             "SPATIAL": lambda self: self._parse_index_constraint(kind="SPATIAL"),
         }
 
+        TABLE_OPTION_PARSERS = {
```

Comment:
I'm wondering if we should make new property classes for these, wdyt @tobymao? I just realized some of them are already supported (e.g. `ALGORITHM` and `CHECKSUM`), so perhaps a `_parse_csv(self._parse_property)` could do the trick here if we created expression types for the unsupported ones.

Reply:
not sure what you mean

Reply:
We can already parse some of these table options, e.g.

https://github.com/tobymao/sqlglot/blob/9595240a1c0f0e5ace9f67f31564e5d5edb9a9d2/tests/dialects/test_mysql.py#L73

The corresponding AST for this is:

```python
>>> import sqlglot
>>> sqlglot.parse_one("CREATE TABLE z (a INT) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARACTER SET=utf8 COLLATE=utf8_bin COMMENT='x'", "mysql")
Create(
  this=Schema(
    this=Table(
      this=Identifier(this=z, quoted=False)),
    expressions=[
      ColumnDef(
        this=Identifier(this=a, quoted=False),
        kind=DataType(this=Type.INT, nested=False))]),
  kind=TABLE,
  properties=Properties(
    expressions=[
      EngineProperty(
        this=Identifier(this=InnoDB, quoted=False)),
      AutoIncrementProperty(
        this=Literal(this=1, is_string=False)),
      CharacterSetProperty(
        this=Var(this=utf8),
        default=True),
      CollateProperty(
        this=Identifier(this=utf8_bin, quoted=False)),
      SchemaCommentProperty(
        this=Literal(this=x, is_string=True))]))
```

I'll see if I can refactor my PR to get rid of the table option parsers and reuse `_parse_properties` or something.

Reply:
https://github.com/tobymao/sqlglot/pull/3027/commits/dba33c04c7d0d03ef3442998cc7159a1df2540c3